src/DataFixtures/DownloadFixtures.php

```diff
@@ -0,0 +1,144 @@
+use Symfony\Component\Console\Output\ConsoleOutput;
+
+/**
+ * Creates fake download statistics for each package.
+ */
+class DownloadFixtures extends Fixture implements DependentFixtureInterface
+{
+    private Client $redis;
+
+    private MigrateDownloadCountsCommand $migrateDownloadCountsCommand;
+
+    public function __construct(Client $redis, MigrateDownloadCountsCommand $command)
+    {
+        $this->redis = $redis;
+        $this->migrateDownloadCountsCommand = $command;
+    }
+
+    public function getDependencies()
+    {
+        return [
+            PackageFixtures::class,
+        ];
+    }
+
+    public function load(ObjectManager $manager)
+    {
+        $input  = new ArrayInput([]);
+        $output = new ConsoleOutput();
+
+        $output->writeln('Generating downloads...');
+
+        /** @var Package[] $packages */
+        $packages = $manager->getRepository(Package::class)->findAll();
+
+        $progressBar = new ProgressBar($output, count($packages));
+        $progressBar->setFormat('%current%/%max% [%bar%] %percent:3s%% (%remaining% left) %message%');
+
+        $progressBar->setMessage('');
+        $progressBar->start();
+
+        // Set the Redis keys that would normally be set by the DownloadManager, for the whole period.
+
+        foreach ($packages as $package) {
+            $progressBar->setMessage($package->getName());
+            $progressBar->display();
+
+            /** @var EntityManagerInterface $manager */
+            $latestVersion = $this->getLatestPackageVersion($manager, $package);
+
+            $this->populateDownloads($package, $latestVersion);
+
+            $progressBar->advance();
+        }
+
+        $progressBar->finish();
+        $output->writeln('');
+
+        // Then migrate the Redis keys to the db
+
+        $output->writeln('Migrating downloads to db... (this may take some time)');
+        $this->migrateDownloadCountsCommand->run($input, $output);
+    }
+
+    /**
+     * Returns the latest non-dev version of the given package.
+     */
+    private function getLatestPackageVersion(EntityManagerInterface $manager, Package $package): Version
+    {
+        return $manager->createQueryBuilder()
+            ->select('v')
+            ->from(Version::class, 'v')
+            ->where('v.package = :package')
+            ->setParameter('package', $package)
+            ->andWhere('v.normalizedVersion NOT LIKE :dev')
+            ->setParameter('dev', 'dev-%')
+            ->orderBy('v.normalizedVersion', 'DESC')
+            ->setMaxResults(1)
+            ->getQuery()
+            ->getSingleResult();
+    }
+
+    /**
+     * Populates Redis with pseudo-random daily download stats starting at the package creation date, and ending today.
+     * The algorithm attempts to mimic a typical download stats curve.
+     * The Redis keys set mimic the keys set by the DownloadManager.
+     */
+    private function populateDownloads(Package $package, Version $version): void
```

Comment:
This only populates redis, then I guess the expectation is that one runs the `packagist:migrate-download-counts` command to move this to the DB? Possibly should be documented.

Reply:
No, it actually runs the command from within the fixtures after populating redis:

https://github.com/BenMorel/packagist/blob/82bb6a141b6e8a14e7a10f7ff091fa485327e298/src/DataFixtures/DownloadFixtures.php#L78-L79